packages/pieces/community/stable-diffusion-webui/src/lib/actions/text-to-image.ts

```diff
@@ -0,0 +1,86 @@
+import { createAction, Property } from '@activepieces/pieces-framework';
+import {
+  httpClient,
+  HttpMethod,
+  HttpRequest,
+} from '@activepieces/pieces-common';
+import { stableDiffusionAuth, StableDiffusionAuthType } from '../../index';
+
+export const textToImage = createAction({
+  name: 'textToImage',
+  displayName: 'Text to Image',
+  description: '',
+  auth: stableDiffusionAuth,
+  props: {
+    prompt: Property.LongText({
+      displayName: 'Prompt',
+      required: true,
+    }),
+    model: Property.Dropdown({
+      displayName: 'Model',
+      required: true,
+      refreshers: ['auth'],
+      options: async ({ auth }) => {
+        if (!auth) {
+          return {
+            disabled: true,
+            options: [],
+            placeholder: 'Please authenticate first',
+          };
+        }
+        const { baseUrl } = auth as StableDiffusionAuthType;
+        const request: HttpRequest = {
+          method: HttpMethod.GET,
+          url: `${baseUrl}/sdapi/v1/sd-models`,
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        };
+        const response = await httpClient.sendRequest(request);
+        const options = response.body
+          ?.map((model: { model_name: string }) => {
+            return {
+              label: model.model_name,
+              value: model.model_name,
+            };
+          })
+          ?.sort((a: { label: string }, b: { label: string }) =>
+            a['label'].localeCompare(b['label'])
+          );
+        return {
+          options: options,
+        };
+      },
+    }),
+    advancedParameters: Property.Object({
+      displayName: 'Advanced parameters (key/value)',
+      required: false,
+      description: 'Refer to API documentation',
+    }),
+  },
+  async run({ auth, propsValue }) {
+    const request: HttpRequest = {
+      method: HttpMethod.POST,
+      url: `${auth.baseUrl}/sdapi/v1/txt2img`,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        ...propsValue.advancedParameters,
+        prompt: propsValue.prompt,
+        override_settings: {
+          sd_model_checkpoint: propsValue.model,
+        },
+        override_settings_restore_afterwards: true,
+      }),
+    };
+    const response = await httpClient.sendRequest(request);
+    return {
+      images: response.body['images']?.map((image: string) => {
+        return {
+          data: `data:image/png;base64,${image}`,
+        };
```

Comment:
We can try to use:

You can give each image a random name or something similar. This will return a local path URL in production mode. When using the test step, it will return a URL that can be viewed by the user instead of base64.
```
        await context.files
          .write({
            fileName: `image.png`,
            data: Buffer.from(artifact.base64, 'base64'),
          })
 ```

Reply:
Done!